src/NetMQ/Core/Patterns/XPub.cs

```diff
@@ -176,15 +184,29 @@ protected override void XReadActivated(Pipe pipe)
                         // If the subscription is not a duplicate, store it so that it can be
                         // passed to used on next recv call.
                         if (m_options.SocketType == ZmqSocketType.Xpub && (unique || m_verbose))
-                            m_pending.Enqueue(sub.CloneData());
+                        {
+                            m_pending.Enqueue(sub);
+                        }
+                        else
+                        {
+                            sub.Close();
+                        }
+
                     }
                 }
+                else if (m_broadcastEnabled && size > 0 && sub.HasMore && sub[0] == 2)
+                {
+                    m_lastPipe = pipe;
+                    m_lastPipeIsBroadcast = true;
+                    sub.Offset = sub.Offset + 1;
+                    sub.Size = sub.Size - 1;
+                    m_pending.Enqueue(sub);
```

Comment:
But I remove prefix only when `sub[0] == 2`, so it doesn't affect sub/unsub messages. The only thing it does is marking the sender (last pipe) for exclusion, but the prefix is not needed for anything else. After these lines, the message should look like a regular one to everyone. The sender just won't receive it back.